### PR TITLE
Fixes #87506 Add apiVersion to involvedObject

### LINKED ISSUE #87506: node Event missing apiVersion in the involvedObject
**What happened**:

I'm attempting to use an `Events` watcher to better re-act to cluster modifications. I've received the below event:

```
{ 
   "metadata":{ 
      "name":"172.26.64.24.15ec9789b5157656",
      "namespace":"default",
      "selfLink":"/api/v1/namespaces/default/events/172.26.64.24.15ec9789b5157656",
      "uid":"28bf0748-3a6e-4590-8c4e-1f262ebb7519",
      "resourceVersion":"107982651",
      "creationTimestamp":"2020-01-23T18:29:49Z"
   },
   "involvedObject":{ 
      "kind":"Node",
      "name":"172.26.64.24",
      "uid":"bb6bd832-895d-11e8-bd5a-001d0917d583"
   },
   "reason":"RegisteredNode",
   "message":"Node 172.26.64.24 event: Registered Node 172.26.64.24 in Controller",
   "source":{ 
      "component":"node-controller"
   },
   "firstTimestamp":"2020-01-23T18:29:47Z",
   "lastTimestamp":"2020-01-23T18:29:47Z",
   "count":1,
   "type":"Normal",
   "eventTime":null,
   "reportingComponent":"",
   "reportingInstance":""
}
```

**What you expected to happen**:

I would expect the `involvedObject` to include `apiVersion` attribute. Not sure if it's all node events or just this particular one.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16
- Cloud provider or hardware configuration: bare metal, rancher's rke
- OS (e.g: `cat /etc/os-release`): centos host nodes
- Kernel (e.g. `uname -a`): 3.18, 4.4
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by uthark**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #87506

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix missing "apiVersion" for "involvedObject" in Events for Nodes.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm
/ok-to-test

please add a short release note explaning this bug fix under "Does this PR introduce a user-facing change?:"

e.g.
> fix missing "apiVersion" for "involvedObject" in Events

**Comment by uthark** [comment]
@neolit123 Done.

File: pkg/controller/util/node/controller_utils.go [hunk 1]
```diff
@@ -159,10 +159,11 @@ func MarkPodsNotReady(kubeClient clientset.Interface, pods []*v1.Pod, nodeName s
 // RecordNodeEvent records a event related to a node.
 func RecordNodeEvent(recorder record.EventRecorder, nodeName, nodeUID, eventtype, reason, event string) {
 	ref := &v1.ObjectReference{
-		Kind:      "Node",
-		Name:      nodeName,
-		UID:       types.UID(nodeUID),
-		Namespace: "",
+		APIVersion: "v1",
```
Comment (mattjmcnaughton) [inline pkg/controller/util/node/controller_utils.go]: I'm not an expert on the k8s api versioning... but just want to check there isn't a risk that the `APIVersion` will increase but we will forget to update this hard coded reference? Is there a constant we can refer to, or method call we can use which will guarantee this value stays up to date?

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for your pr :)

Up to you if you want to add this... but should we add/update a unit test to verify the `APIVersion` is included?

**Comment by uthark** [comment]
Good point. I'll update PR.

Best regards, Oleg.

**Comment by khyveasna11111908** [comment]
Has a little better but still not received

**Review by khyveasna11111908 — APPROVED** [review]
Garcia and my dad is

File: pkg/controller/util/node/controller_utils.go (hunk 1 above)
Comment (uthark) [reply pkg/controller/util/node/controller_utils.go]: I searched over the codebase and I see that often APIVersion is omitted.

Comment (rphillips) [reply pkg/controller/util/node/controller_utils.go]: This is sorta related to the nodeRef not being a real node object in the Kubelet. I think this patch is ok, but it definitely should be refactored at some point.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L531-L533

**Pull request merged**